src/sentry/api/event_search.py

```diff
@@ -405,41 +409,41 @@ def process_list(self, first, remaining):
         ]
 
     def visit_numeric_filter(self, node, children):
-        (search_key, _, (value,)) = children
-        operator = value[0]
+        (search_key, _, operator, search_value) = children
         if isinstance(operator, Node):
-            if isinstance(operator.expr, Optional):
-                operator = "="
-            else:
-                operator = operator.text
+            operator = "=" if isinstance(operator.expr, Optional) else operator.text
         else:
             operator = operator[0]
 
-        if operator == "[":
-            operator = "IN"
-            search_value = self.process_list(value[1], value[2])
+        if self.is_numeric_key(search_key.name):
+            try:
+                search_value = SearchValue(parse_numeric_value(*search_value.match.groups()))
+            except InvalidQuery as exc:
+                raise InvalidSearchQuery(str(exc))
+            return SearchFilter(search_key, operator, search_value)
         else:
-            search_value = value[1]
+            search_value = search_value.text
+            search_value = SearchValue(operator + search_value if operator != "=" else search_value)
+            operator = "=" if operator != "=" else operator
+
+            return self._handle_basic_filter(search_key, operator, search_value)
+
+    def visit_numeric_in_filter(self, node, children):
+        (search_key, _, value) = children
+        operator = "IN"
+        search_value = self.process_list(value[1], value[2])
 
         if self.is_numeric_key(search_key.name):
             try:
                 search_value = SearchValue(
                     [parse_numeric_value(*val.match.groups()) for val in search_value]
-                    if operator == "IN"
-                    else parse_numeric_value(*search_value.match.groups())
                 )
             except InvalidQuery as exc:
                 raise InvalidSearchQuery(str(exc))
             return SearchFilter(search_key, operator, search_value)
         else:
-            if operator != "IN":
-                search_value = search_value.text
-            else:
-                search_value = [v.text for v in search_value]
-            search_value = SearchValue(
-                operator + search_value if operator not in ("=", "IN") else search_value
-            )
-            operator = "=" if operator not in ("=", "IN") else operator
+            search_value = [v.text for v in search_value]
+            search_value = SearchValue(search_value)
             return self._handle_basic_filter(search_key, operator, search_value)
```

Comment:
Honestly the diff here is pretty hard to read, it might be easier to just look at the file with the changes are applied.